### PR TITLE
[ParseableInterfaces] Print @_show_in_interface

### PR DESCRIPTION
**Pull request by harlanhaskins**
This attribute needs to be preserved in the .swiftmodule, otherwise these declarations will stop showing up in the interface. Print it in the parseable interface.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

File: lib/AST/Attr.cpp [hunk 1]
```diff
@@ -374,10 +374,13 @@ bool DeclAttribute::printImpl(ASTPrinter &Printer, const PrintOptions &Options,
   case DAK_RawDocComment:
   case DAK_ObjCBridged:
   case DAK_SynthesizedProtocol:
-  case DAK_ShowInInterface:
   case DAK_Rethrows:
   case DAK_Infix:
     return false;
+  case DAK_ShowInInterface:
+    if (!Options.PrintShowInInterfaceAttr)
+      return false;
```
Comment (jrose-apple) [inline lib/AST/Attr.cpp]: Maybe just have this in the ExcludeAttrList by default instead?

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Review by jrose-apple — APPROVED** [review]

**Review by nathawes — APPROVED** [review]

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Pull request merged**